modules/objdetect/include/opencv2/objdetect/aruco_board.hpp

```diff
@@ -0,0 +1,301 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html
+#ifndef __OPENCV_ARUCO_BOARD_HPP__
+#define __OPENCV_ARUCO_BOARD_HPP__
+
+#include <opencv2/calib3d.hpp>
+#include <opencv2/core.hpp>
+#include <vector>
+
+namespace cv {
+namespace aruco {
+//! @addtogroup aruco
+//! @{
+
+class Dictionary;
+
+/**
+ * @brief Board of markers
+ *
+ * A board is a set of markers in the 3D space with a common coordinate system.
+ * The common form of a board of marker is a planar (2D) board, however any 3D layout can be used.
+ * A Board object is composed by:
+ * - The object points of the marker corners, i.e. their coordinates respect to the board system.
+ * - The dictionary which indicates the type of markers of the board
+ * - The identifier of all the markers in the board.
+ */
+class CV_EXPORTS_W Board {
+public:
+    CV_WRAP Board();
+
+    /**
+     * @brief Draw a planar board
+     *
+     * @param outSize size of the output image in pixels.
+     * @param img output image with the board. The size of this image will be outSize
+     * and the board will be on the center, keeping the board proportions.
+     * @param marginSize minimum margins (in pixels) of the board in the output image
+     * @param borderBits width of the marker borders.
+     *
+     * This function return the image of the GridBoard, ready to be printed.
+     */
+    CV_WRAP virtual void draw(Size outSize, OutputArray img, int marginSize = 0, int borderBits = 1) const;
+
+    /** @brief Provide way to create Board by passing necessary data. Specially needed in Python.
+     * @param objPoints array of object points of all the marker corners in the board
+     * @param dictionary the dictionary of markers employed for this board
+     * @param ids vector of the identifiers of the markers in the board
+     */
+    CV_WRAP static Ptr<Board> create(InputArrayOfArrays objPoints, const Ptr<Dictionary> &dictionary, InputArray ids);
+
+    /** @brief Set ids vector
+     * @param ids vector of the identifiers of the markers in the board (should be the same size
+     * as objPoints)
+     *
+     * Recommended way to set ids vector, which will fail if the size of ids does not match size
+     * of objPoints.
+     */
+    CV_WRAP void setIds(InputArray ids);
+
+    /** @brief change id for ids[index]
+     * @param index - element index in ids
+     * @param newId - new value for ids[index], should be less than Dictionary size
+     */
+    CV_WRAP void changeId(int index, int newId);
+    /** @brief return ids
+     */
+    CV_WRAP const std::vector<int>& getIds() const;
+
+    /** @brief set dictionary
+     */
+    CV_WRAP void setDictionary(const Ptr<Dictionary> &dictionary);
```

Comment:
Could `.setDictionary()` call break instance? (make it inconsistent due to missing `.setIds()` call)

P.S. We don't need API which allows to "shoot legs".